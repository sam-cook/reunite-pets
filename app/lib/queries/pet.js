import gql from "graphql-tag";

export const petById = gql`
  query Pet($id: UUID!) {
    petById(id: $id) {
      id
      breed
      name
      species
    }
  }
`;
