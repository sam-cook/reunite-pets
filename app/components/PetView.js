import React from "react";
import { Query } from "react-apollo";
import ErrorMessage from "./ErrorMessage";
import { petById } from "../lib/queries/pet";

export default function PetView(props) {
  return (
    <Query query={petById} variables={{ id: props.id }}>
      {({ loading, error, data }) => {
        if (error) return <ErrorMessage message="Error loading pet :-(." />;
        if (loading) return <div>Loading</div>;
        const pet = data.petById;
        return (
          <section>
            <h1>
              {pet.name} - {pet.breed}
            </h1>
          </section>
        );
      }}
    </Query>
  );
}
