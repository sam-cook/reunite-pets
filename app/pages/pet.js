import React, { Component } from "react";
import PropTypes from "prop-types";
import PetView from "../components/PetView";

export default class Pet extends Component {
  static propTypes = {
    req: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string
      }).isRequired
    }).isRequired,
    id: PropTypes.string
  };

  static defaultProps = {
    id: null
  };

  static getInitialProps(props) {
    return {
      id: props.req.params.id
    };
  }

  render() {
    return <PetView id={this.props.id} />;
  }
}
