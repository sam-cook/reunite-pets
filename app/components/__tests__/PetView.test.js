import React from "react";
import wait from "waait";
import { MockedProvider } from "react-apollo/test-utils";
import { render } from "@testing-library/react";
import { petById } from "../../lib/queries/pet";
import PetView from "../PetView";

const mocks = [
  {
    request: {
      query: petById,
      variables: {
        id: "wew lad"
      }
    },
    result: {
      data: {
        petById: {
          id: "wew lad",
          name: "Buck",
          breed: "bulldog",
          species: "dog"
        }
      }
    }
  }
];

it("renders a component or something", async () => {
  const { container } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <PetView id="wew lad" />
    </MockedProvider>
  );

  await wait(0);

  expect(container.firstChild).toMatchSnapshot();
});
