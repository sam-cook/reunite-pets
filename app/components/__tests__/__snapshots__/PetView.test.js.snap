// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a component 1`] = `
<section>
  <h1>
    Buck
     - 
    bulldog
  </h1>
</section>
`;
